src/script/lua_api/l_object.cpp

```diff
@@ -2539,7 +2539,26 @@ int ObjectRef::l_set_lighting(lua_State *L)
 			lighting.volumetric_light_strength = rangelim(lighting.volumetric_light_strength, 0.0f, 1.0f);
 		}
 		lua_pop(L, 1); // volumetric_light
-}
+
+		lua_getfield(L, 2, "light_intensity");
+		if (lua_istable(L, -1)) {
+			lua_getfield(L, 3, "color_offset");
+			if (lua_istable(L, -1)) {
+				lighting.lightIntensity.colorOffset_rgb.X = getfloatfield_default(L, -1, "r", lighting.lightIntensity.colorOffset_rgb.X);
```

Comment:
Create a some reference to the structure like `LightIntensity &sky_light = lighting.lightIntensity` to avoid writing overly long lines. Also, I think it is worth to remove `_rgb` suffix since it is pretty clear this is RGB format by judging the storing only 3 float values (v3f).